helios-client/src/main/java/com/spotify/helios/client/RetryingRequestDispatcher.java

```diff
@@ -120,22 +118,72 @@ public void onFailure(@NotNull Throwable t) {
   private void handleFailure(final SettableFuture<Response> future,
                                  final Supplier<ListenableFuture<Response>> code,
                                  final long deadline,
-                                 final long delay,
-                                 final TimeUnit timeUnit,
+                                 final long delayMillis,
                                  final Throwable t) {
     if (clock.now().getMillis() < deadline) {
-      if (delay > 0) {
+      if (delayMillis > 0) {
         executorService.schedule(new Runnable() {
           @Override
           public void run() {
-            startRetry(future, code, deadline - 1, delay, timeUnit);
+            startRetry(future, code, deadline - 1, delayMillis);
           }
-        }, delay, timeUnit);
+        }, delayMillis, TimeUnit.MILLISECONDS);
       } else {
-        startRetry(future, code, deadline - 1, delay, timeUnit);
+        startRetry(future, code, deadline - 1, delayMillis);
       }
     } else {
       future.setException(t);
     }
   }
+
+  static Builder forDispatcher(RequestDispatcher delegate) {
```

Comment:
I'm slightly curious why you call this `forDispatcher` as opposed to just `builder`